solidity/contracts/utility/ContractRegistryClient.sol

```diff
@@ -0,0 +1,110 @@
+pragma solidity 0.4.26;
+import './Owned.sol';
+import './Utils.sol';
+import './interfaces/IContractRegistry.sol';
+
+/**
+  * @dev Base contract for ContractRegistry clients
+*/
+contract ContractRegistryClient is Owned, Utils {
+    IContractRegistry public registry;      // address of the current contract-registry
+    IContractRegistry public prevRegistry;  // address of the previous contract-registry
+    bool public allowRegistryUpdate = true; // allow/prevent changing the contract-registry
+
+    /**
+      * @dev initializes a new ContractRegistryClient instance
+      * 
+      * @param  _registry   address of a contract-registry contract
+    */
+    constructor(IContractRegistry _registry) internal validAddress(_registry) {
+        registry = IContractRegistry(_registry);
+        prevRegistry = IContractRegistry(_registry);
+    }
+
+    /**
+      * @dev returns whether or not the caller is authorized to change the contract-registry
+     */
+    function callerIsAuthorized() internal view returns (bool) {
+        return msg.sender == owner;
+    }
+
+    /**
+      * @dev sets the contract-registry to the address of the contract-registry in the contract-registry
+     */
+    function updateRegistry() public {
+        // verify that either updating is allowed or the caller is authorized
+        require(allowRegistryUpdate || callerIsAuthorized());
+
+        // get the address of the contract-registry in the contract-registry
+        address newRegistry = addressOf(CONTRACT_REGISTRY);
+
+        // verify that the new contract-registry is different and not zero
+        require(newRegistry != address(registry) && newRegistry != address(0));
+
+        // set the previous contract-registry as current contract-registry
+        prevRegistry = registry;
+
+        // set the current contract-registry as the new contract-registry
+        registry = IContractRegistry(newRegistry);
+    }
+
+    /**
+      * @dev reverts back to the previous contract-registry
+    */
+    function restoreRegistry() public {
+        // verify that either restoring is allowed or the caller is authorized
+        require(allowRegistryUpdate || callerIsAuthorized());
+
+        // set the current contract-registry as the previous contract-registry
+        registry = prevRegistry;
+
+        // ensure that only an authorized caller can perform future changes
+        allowRegistryUpdate = false;
+    }
+
+    /**
+      * @dev allows/prevents changing the contract-registry
+      * 
+      * @param _disable    true to disable changes, false to enable changes
+    */
+    function disableRegistryUpdate(bool _disable) public ownerOnly {

```

Comment:
disable -> restrict?

Reply:
Yeah, true. Well, it seems that we'll need to revise our entire terminology in this contract.
It's also gonna be an API-breaking change, because I guess that it won't be much of an issue since all of the functionality is generally restricted to privileged users, right?